akirachix/Kishya-Frontend
# added web portal

What this PR does?
This PR enhances the MapDisplay component by introducing several improvements and optimizations, particularly related to the handling of location data and user interactions. The following key changes have been made:

Optional mapInfo Property: The mapInfo property in the FloodRiskData interface is now optional, allowing for more flexibility when handling location data.
Feedback Form Visibility: Improved management of the feedback form visibility state to enhance user experience.
Dynamic Data Fetching: Enhanced the functionality for fetching coordinates and location-specific data when a user performs a search.
UI Element Management: Streamlined the hiding and showing of UI elements when downloading the PDF report.
Type of Change
 Improvement
 Bug fix
Motivation and Context
The motivation behind these changes is to create a more robust and user-friendly map interface for displaying flood risk information. By making mapInfo optional, we ensure that the application can gracefully handle cases where certain data may not be available. Additionally, improving the dynamic data-fetching logic ensures that users receive accurate and timely information based on their search inputs.

Dependencies
html2canvas for capturing the map as an image
jsPDF for generating PDF reports
react-toastify for user notifications
How can it be tested?
Check Optional mapInfo:

Ensure that the application can function correctly even if mapInfo is not provided in the data.
Search Functionality:

Enter a location in the search bar and verify that the map updates to center on the new location and that the side panel data is correctly populated.
Feedback Form:

Open and close the feedback form to ensure that it displays and hides properly.
PDF Download:

Perform a search, then attempt to download the PDF report. Confirm that the UI elements are hidden during the download process and that the report includes the correct information.


Responsive Design:
![image](https://github.com/user-attachments/assets/c83715ba-78d1-40ad-94de-abf0bf7477bf)

Test the application on different screen sizes to ensure that the layout adapts appropriately.
# Where is it documented?
- List JIRA tickets
